src/baldr/graphreader.cc

```diff
@@ -203,39 +166,22 @@ TileCache* TileCacheFactory::createTileCache(const boost::property_tree::ptree&
 }
 
 // Constructor using separate tile files
-GraphReader::GraphReader(const boost::property_tree::ptree& pt)
-    : tile_hierarchy_(pt.get<std::string>("tile_dir")),
-      tile_extract_(get_extract_instance(pt)),
+GraphReader::GraphReader(const std::shared_ptr<GraphTileStorage>& tile_storage, const boost::property_tree::ptree& pt)
```

Comment:
since the `tile_storage` is the only thing that uses the `ptree` anymore can you please remove the `ptree` from the `GraphReader` interface?